### ISSUE TITLE
Line Not Full Width on Composed Chart with Bars

### ISSUE THREAD

**Issue by AndyDyer** [recharts/recharts#2615]
- [ ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/competent-austin-r9ycq?file=/src/App.js)

### Steps to reproduce
Add A Line to a composed chart with bars

### What is expected?
I expect or want to be able to make the line full width. I expect the trend line to extend to the left-most part of the left-most bar and the right-most part of the right-most bar? at least optionally.

### What is actually happening?
instead it is gated by the bars,  

I understand this could be considered a misrepresentation of the data but i still want to force this.

![Screen Shot 2021-07-29 at 12 19 14 PM](https://user-images.githubusercontent.com/22456292/127536429-e76e948f-d116-41f5-862e-ecc363d5220f.png)

| Environment | Info |
|---|---|
| Recharts | v2.0.3 |
| React | 16.13 |
| System | OSX Big Sur |
| Browser | Chrome 92 |

**Comment by ckifer** [comment]
Closing this in favor of other issues repressing the same thing

**Comment by dwolrdcojp** [comment]
If anyone is looking for a work around here is a way to achieve this. This example is for a value that needs to plot as a straight line so depending on your data set you may need to update this code to fit. 

```ts
type YScale = (value: number) => number;

type YAxisMap = {
  [key: number]: {
    scale: YScale;
  };
};

type ChartOffset = {
  left: number;
  right: number;
  top: number;
  bottom: number;
  width: number;
  height: number;
};

type Props = {
  value?: number;
  yAxisMap?: YAxisMap;
  offset?: ChartOffset;
};

export function FullWidthLine({
  value,
  yAxisMap,
  offset,
}: Props) {
  const yScale = yAxisMap?.[0]?.scale;
  if (!yScale || !offset) return null;

  const x1 = offset.left;
  const x2 = offset.width + offset.left;

  return (
    <g>
      {value != null && (
        <line
          x1={x1}
          x2={x2}
          y1={yScale(value)}
          y2={yScale(value)}
          stroke="hsla(var(--value))"
          strokeWidth={1.5}
        />
      )}
    </g>
  );
}
```

In your `<ComposedChart />` children:

```ts
        <Customized
          component={(props: any) => (
            <FullWidthLine
              {...props}
              value={value}
            />
          )}
        />
```
Result: 
<img width="2840" height="820" alt="Image" src="https://github.com/user-attachments/assets/2770f05b-fe65-48e6-82a6-cc4f8043e58c" />
